File: my_py_pkg/my_py_pkg/my_first_node.py
import rclpy
from rclpy.node import Node

class MyNode(Node):
    def __init__(self):
        super().__init__('py_test')
        self.create_timer(2, self.timer_callback)

    def timer_callback(self):
        self.get_logger().info('Python is so great :D')


def main(args=None):
    rclpy.init(args=args)

    node = MyNode()

    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: my_cpp_pkg/src/my_first_node.cpp
#include "rclcpp/rclcpp.hpp"

class MyNode: public rclcpp::Node {
    public:

    MyNode(): rclcpp::Node("cpp_test"), counter(0) {
        RCLCPP_INFO(this->get_logger(), "bruh2");

        timer_ = this->create_wall_timer(std::chrono::seconds(2), std::bind(&MyNode::timerCallback, this));
    }

    private:
    
    void timerCallback() {
        RCLCPP_INFO(this->get_logger(), "Hello %d",++counter);
    }

    int counter = 0;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char** argv) {
    rclcpp::init(argc, argv);

        auto node = std::make_shared<MyNode>();
    
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
}